aioscheduler/scheduler.py

```diff
@@ -83,10 +83,10 @@ async def loop(self) -> None:
             # Sleep until task will be executed
             done, _ = await asyncio.wait(
```

Comment:
This won't actually cancel the pending task, you should definitely do that manually.
```py
done, pending = await asyncio.wait(...)
for task in pending:
    task.cancel()
```

Reply:
I’ll do that but that is how the code was originally lol

Reply:
Futures (as used in the original code) don't need to be cancelled, your modifications create tasks which have to be cancelled.

Reply:
Ah that makes sense